ocean-client/addons/command-center/src/command-center.ts

```diff
@@ -0,0 +1,100 @@
+import { Skip } from './commands/skip'
+import { checkSafetyOf } from './utils/helpers'
+import { Store, StoredSettings } from './utils/store'
+import { Message, Telegram } from './utils/telegram'
+
+const VERSION = "v1.0beta"
+
+async function execute(messages: Message[], telegram: Telegram, store: Store) {
+    for (const message of messages) {
+        let commandData = message.command.split(" ")
+        if (commandData.length == 0) {
+            continue
+        }
+        let command: Command | undefined
+        switch (commandData[0]) {
+            case Commands.Help:
+                command = new Help(telegram)
+                break
+            case Commands.CheckMaxi:
+                command = new CheckMaxi(telegram)
+                break
+            case Commands.Execute:
+                command = new Execute(telegram)
+                break
+            case Commands.Skip:
+                command = new Skip(telegram, store)
+                break
+            case Commands.Resume:
+                command = new Resume(telegram, store)
+                break
+            case Commands.RemoveExposure:
+                command = new RemoveExposure(telegram, store)
+                break
+            case Commands.SetRange:
+                command = new SetRange(telegram, store, commandData)
+                break
+            case Commands.SetReinvest:
+                command = new SetReinvest(telegram, store, commandData)
+                break
+            case Commands.SetStableArbSize:
+                command = new SetStableArbSize(telegram, store, commandData)
+                break
+            case Commands.SetAutoDonation:
+                command = new SetAutoDonation(telegram, store, commandData)
+                break
+            default:
+                console.log("ignore " + message.command)
+                break
+        }
+        await telegram.send("executing " + message.command)
+        await command?.execute()
+    }
+}
+
+function isFirstRun(settings: StoredSettings): boolean {
+    return settings.lastExecutedMessageId === 0
+}
+
+export async function main(): Promise<Object> {
+
+    const store = new Store()
+    const settings = await store.fetchSettings()
+
+    const telegram = new Telegram(settings, "[CommandCenter " + process.env.AWS_REGION + " " + VERSION + "]")
+
+    let messages = await telegram.getMessages()
+    messages = messages.filter((message) => {
+        return checkSafetyOf(message, settings)
+    })
+    if (messages.length > 0) {
+        // Krysh: update last message right away to avoid infinite error loops because of some command
+        // couldn't be processed.
+        await store.updateExecutedMessageId(messages[messages.length - 1].id)
+        console.log("messages to be executed", messages.length)
+        // Krysh: if we are on the first run, ignore every command besides the last one
+        // to prevent any "old" commands to be executed
```

Comment:
why even consider the last one?
probably better to ignore any message on the first run. better safe than sorry.

Reply:
Thought the same thing, but in 99.9% of the cases, there will be no chat with commands and there the very first message - will trigger the whole thing and therefore reduce it.

I think it is the best solution to avoid multiple execution of old commands and to execute the first command of a new user